### DISCUSSION TITLE
Pipeline operator (tweet)

### DISCUSSION THREAD

**Discussion by ggreif** [dfinity/motoko#4835] [Show and tell]
Many languages provide a way to chain function calls by invoking methods on the result of the previous call. This is called [fluent syntax](https://en.wikipedia.org/wiki/Fluent_interface). In Motoko such a practice would have bad performance characteristics, so we have the pipeline operator instead:
``` Motoko
getTable("alumni") |> filter (_, func alumnus = alumnus.year == 2004) |> print(_[0].name)
```
Here the special binding `_` stands for the value originating from the preceding pipeline stage.

**Comment by jessiemongeon1** [comment]
Scheduled for 4/25
